diff --git a/node_modules/ai/dist/index.mjs b/node_modules/ai/dist/index.mjs
index dd26356..892db29 100644
--- a/node_modules/ai/dist/index.mjs
+++ b/node_modules/ai/dist/index.mjs
@@ -1719,7 +1719,16 @@ function createEventStreamTransformer(customParser) {
       );
     },
     transform(chunk) {
-      eventSourceParser.feed(textDecoder.decode(chunk));
+      // patch-package start
+      // Support Ollama + Llava stream
+      // eventSourceParser.feed(textDecoder.decode(chunk));
+      const event = textDecoder.decode(chunk);
+      if (event.startsWith("data: ")) {
+        eventSourceParser.feed(event);
+      } else {
+        eventSourceParser.feed(`data: ${event}\n`);
+      }
+      // patch-package end
     }
   });
 }
